platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonModelWriter.kt

```diff
@@ -37,15 +35,11 @@ enum class DiagnosticKind {
     INCOMPATIBLE_TASK
 }
 
-
-class JsonModelWriter(val writer: Writer) {
+class JsonModelWriter(writer: Writer): JsonModelWriterCommon(writer) {
```

Comment:
❌  Since the new writer is a general utility to write JSON's, it should be used via composition rather than inheritance